scrapy/exporters.py

```diff
@@ -71,7 +76,10 @@ def _get_serialized_fields(self, item, default_value=None, include_empty=None):
 
         for field_name in field_iter:
             if field_name in item:
-                field = {} if isinstance(item, dict) else item.fields[field_name]
+                try:
+                    field = item.get_field_meta(field_name)
+                except TypeError:
+                    field = {}
```

Comment:
Are there cases when `item.get_field_meta(field_name)` for an item type which doesn't support meta should return anything other than `{}`?

Reply:
I'm not sure I understand the question, but I see how using `TypeError` could be bad. Adding a `default` parameter sounds coherent with `dict.get`. I'll make the changes in itemadapter.

Reply:
See https://github.com/scrapy/itemadapter/pull/19

Reply:
`item.get_field_meta` currently raises an exception if an item doesn't provide a way to set meta. Now I think that there should be a way for it not to raise an exception. What's the use case for this exception?

If there is no use case, we should just return `{}`. There is no use case for an exception in Scrapy itself, as in all Scrapy usages exception handling won't be needed if a `{}` default is returned when meta is absent.

If there is a possibility that there is a use case, but it is not clear which use case is that, it'd be good to keep an option to have this exception, but make the lookup return `{}` by default, to match the common usage better. 

If there is a use case, then we should consider making exception a default behavior.

For me returning empty dict as a field meta for dicts (or other future possibile items which don't provide meta) feels ok. Meta is optional in dataclasses and attrs, so it is normal behavior that the meta is not populated. As it is optional, that's not a problem that for dict meta is empty - it could be empty for attrs or dataclasses as well. The difference is that dict doesn't allow to set meta, but that's user choice. If they realize they need to set meta, they switch to scrapy Items or attrs or dataclasses. It is actually similar for attrs and dataclasses: you can't set meta if you use type annotation-based definitions; if user realizes they need meta, they switch to e.g. to an explicit attr.ib(...).

Reply:
Updated in a3672d5e81aeec76b6c5718dd9f9f61f43edf1f3, https://github.com/scrapy/itemadapter/commit/3594f8f52767e499c86182e25509dce893cbf6ae